pkg/common/utils/utils.go

```diff
@@ -121,8 +152,45 @@ func QuerySnapshotsUtil(ctx context.Context, m cnsvolume.Manager,
 		if len(snapshotQueryResult.Entries) == 0 {
 			log.Infof("QuerySnapshots retrieved no results for the spec: %+v", snapshotQuerySpec)
 		}
+		// Update the max iteration.
+		// maxIterationSet ensures that the max iterations are set only once, this is to ensure that the number of
+		// results are lower than the max entries supported by caller in a busy system which has increasing number
+		// total records.
+		if !maxIterationSet {
+			if snapshotQueryResult.Cursor.TotalRecords < maxEntries {
+				// If the total number of records is lesser than max entries supported by caller then
+				// all results can be retrieved in a loop, when the results are returned no next-token is expected to be set.
+				// Example:
+				// maxEntries=200, totalRecords=150, batchSize=128
+				// maxIteration=2
+				// iteration-1: 128 results, iteration-2: 22 results
+				// total results returned: 150
+				// offset=0
+				maxRecords := snapshotQueryResult.Cursor.TotalRecords
+				numOfIterationsForAllResults := float64(maxRecords) / float64(batchSize)
+				maxIteration = int64(math.Ceil(numOfIterationsForAllResults))
+				log.Infof("setting max iteration to %d for total records count: %d", maxIteration, maxRecords)
+				maxIterationSet = true
```

Comment:
These 4 lines and L184~188 are the same. You can extra them out.

Reply:
```
maxRecords := snapshotQueryResult.Cursor.TotalRecords
numOfIterationsForAllResults := float64(maxRecords) / float64(batchSize)
maxIteration = int64(math.Ceil(numOfIterationsForAllResults))
```

```
maxRecords := maxEntries
numOfIterationsForAllResults := float64(maxRecords) / float64(batchSize)
maxIteration = int64(math.Floor(numOfIterationsForAllResults))
```
Both are quite different.

Reply:
OK. You can just extra the following line out.
```
maxIterationSet = true
```

Reply:
done.